tests/raster/test_geotiff.py

```diff
@@ -0,0 +1,307 @@
+        """
+        Test write and read of a multi band mosaic.
+        """
+        data = np.ones((5, 100, 100), dtype=np.float32)
+
+        with GeoTiffFile(self.filepath, mode='w', n_bands=5) as src:
+            src.write(data)
+
+        with GeoTiffFile(self.filepath) as src:
+            for band in np.arange(data.shape[0]):
+                ds = src.read(bands=band + 1)
+                np.testing.assert_array_equal(
+                    ds[band + 1], data[band, :, :])
+
+    def test_decoding_multi_band(self):
+        """
+        Test decoding of multi band mosaic.
+
+        """
+        data = np.ones((5, 100, 100), dtype=np.float32)
+        scale_factors = {1: 1, 2: 2, 3: 1, 4: 1, 5: 3}
+        offsets = {2: 3}
+        with GeoTiffFile(self.filepath, mode='w', n_bands=5,
+                         scale_factors=scale_factors, offsets=offsets) as src:
+            src.write(data)
+
+        with GeoTiffFile(self.filepath, auto_decode=False) as src:
+            ds = src.read(bands=1)
+            np.testing.assert_array_equal(ds[1], data[0, :, :])
+            ds = src.read(bands=2)
+            np.testing.assert_array_equal(ds[2], data[1, :, :])
+            ds = src.read(bands=5)
+            np.testing.assert_array_equal(ds[5], data[4, :, :])
+
+        data[1, :, :] = data[1, :, :] * 2 + 3
+        data[4, :, :] = data[4, :, :] * 3
+
+        with GeoTiffFile(self.filepath, auto_decode=True) as src:
+            ds = src.read(bands=1)
+            np.testing.assert_array_equal(ds[1], data[0, :, :])
+            ds = src.read(bands=2)
+            np.testing.assert_array_equal(ds[2], data[1, :, :])
+            ds = src.read(bands=5)
+            np.testing.assert_array_equal(ds[5], data[4, :, :])
+
+    def test_read_write_specific_band(self):
+        """
+        Test a write and read of a specific mosaic.
+        """
+        data = np.ones((100, 100), dtype=np.float32)
+
+        with GeoTiffFile(self.filepath, mode='w', n_bands=10) as src:
+            src.write({5: data})
+            src.write({10: data})
+
+        with GeoTiffFile(self.filepath) as src:
+            ds = src.read(bands=5)
+            np.testing.assert_array_equal(ds[5], data)
+            ds = src.read(bands=10)
+            np.testing.assert_array_equal(ds[10], data)
+
+    def test_metadata(self):
+        """
+        Test meta mosaic tags.
+        """
+        data = np.ones((1, 100, 100), dtype=np.float32)
+
+        metadata = {'attr1': '123', 'attr2': 'test'}
+
+        with GeoTiffFile(self.filepath, mode='w', metadata=metadata) as src:
+            src.write(data)
+
+        with GeoTiffFile(self.filepath) as src:
+            ds_md = src.metadata
+
+        self.assertEqual(ds_md, metadata)
+
+    def test_geotransform(self):
+        """
+        Test geotransform and spatialref keywords.
+
+        Notes
+        -----
+        Spatial reference cannot be compared due to different versions of WKT1 strings.
+
+        """
```

Comment:
Don't write this in a doc string, write the test function more expressively instead, so it is immediatly clear when running the test framework what exactly went wrong. i.e. test_correct_serialization_of_geo_transformation_and_spatial_references

Reply:
Done.